packages/fiori-freestyle/templates/worklist/add/webapp/controller/Object.controller.js

```diff
@@ -19,37 +20,34 @@ sap.ui.define([
 		 */
 		onInit : function () {
 			// Model used to manipulate control states. The chosen values make sure,
-			// detail page is busy indication immediately so there is no break in
+			// detail page shows busy indication immediately so there is no break in
 			// between the busy indication for loading the view's meta data
-			var iOriginalBusyDelay,
-				oViewModel = new JSONModel({
+			var oViewModel = new JSONModel({
 					busy : true,
 					delay : 0
 				});
-
 			this.getRouter().getRoute("object").attachPatternMatched(this._onObjectMatched, this);
-
-			// Store original busy indicator delay, so it can be restored later on
-			iOriginalBusyDelay = this.getView().getBusyIndicatorDelay();
 			this.setModel(oViewModel, "objectView");
-			this.getOwnerComponent().getModel().metadataLoaded().then(function () {
-					// Restore original busy indicator delay for the object view
-					oViewModel.setProperty("/delay", iOriginalBusyDelay);
-				}
-			);
 		},
-
 		/* =========================================================== */
 		/* event handlers                                              */
 		/* =========================================================== */
 
 
 		/**
-		 * Event handler for navigating back.
+		 * Event handler  for navigating back.
+		 * It there is a history entry we go one step back in the browser history
+		 * If not, it will replace the current entry of the browser history with the worklist route.
 		 * @public
 		 */
 		onNavBack : function() {
-			this.getRouter().navTo("worklist", {}, true);
+			var sPreviousHash = History.getInstance().getPreviousHash();
+			if (sPreviousHash !== undefined) {
+				// eslint-disable-next-line sap-no-history-manipulation
+				history.go(-1);
+			} else {
+				this.getRouter().navTo("worklist", {}, true);
```

Comment:
Is it correct that in one case we use the Router for navigation and in the other we directly manipulating the history? Couldn't we use the router for both?

Reply:
I think this is correct as the router doesn't provide the abstraction necessary for this.
In case there is already a history stack this means that we come from the worklist so our history looks like
`[worklist, detailPage]`
At that point doing a navTo without replace context would have us end up with
`[worklist, detailPage, worklist]` which is not what we want
Doing a navTo with replace would end up with
`[worklist, workslist]` which is also not what we want.

So history.go(-1) is the only way to get to the right state.